pallets/aliases/src/lib.rs

```diff
@@ -0,0 +1,379 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+extern crate core;
+
+use codec::{Compact, Decode, Encode};
+use frame_support::traits::tokens::fungible;
+pub use pallet::*;
+use pallet_clusters::Clusters;
+use pallet_fragments::GetInstanceOwnerParams;
+use pallet_protos::{Proto, ProtoOwner, Protos};
+use sp_clamor::{Hash128, Hash256, InstanceUnit};
+use sp_std::vec::Vec;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(test)]
+mod dummy_data;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+/// Enum that indicates the types of target assets linked to an alias
+#[derive(Encode, Decode, Clone, scale_info::TypeInfo, Debug, PartialEq)]
+pub enum LinkTarget<TAccountId> {
+	Proto(Hash256),
+	Fragment { definition_hash: Vec<u8>, edition_id: InstanceUnit, copy_id: InstanceUnit }, // struct variant allows a nicer UX on polkadotJS
```

Comment:
Why `Vec<u8>` here?

Reply:
Because it is how it is set in `GetInstanceOwnerParams` struct in `pallet-fragments`, which I am reusing here to check the ownership of a fragment instance (by calling the related function in pallet-fragments).

Reply:
`GetInstanceOwnerParams` is used in RPCs, should be unrelated with storage? You can translate types when going into RPCs world.

Reply:
It is unrelated to storage, I do not store this struct in `pallet-aliases`. Since here I need to check ownership of the Fragment instance when creating the alias, reusing this struct available already in `pallet-fragment` made it easy to reuse also the same function from `pallet-fragment` that does exactly what I need here. Indeed, in this PR I also quickly refactored the function `get_instance_owner` called by the RPC in `pallet-fragment` , where I extracted the part that makes the ownership check in another function called `get_instance_owner_account_id` that returns an `AccountId`. The RPC function `get_instance_owner` calls that function and it does what it was doing before, i.e. encoding it to return a `Vec<u8>`. This way, we also have a reusable internal function in `pallet-fragment` that can be reused by other pallets if we need to check ownership of Fragment instance.
Can this work?

Reply:
Done.